alacritty_terminal/src/term/mod.rs

```diff
@@ -687,6 +687,87 @@ impl VisualBell {
     }
 }
 
+/// LineDamage describes the current damage of a line, represented as two
+/// column coordinates describing the edges of the damage. Damage is valid if
+/// left <= right.
+#[derive(Clone)]
+pub struct LineDamage {
+    pub left: Column,
+    pub right: Column,
+    pub line: Line,
+}
+
+impl LineDamage {
+    /// Create a new invalid LineDamage, which has left=columns+1 and right=0.
+    /// This ensures that later update with min/max of any valid column
+    /// automatically creates a valid line damage without needing any
+    /// conditionals.
+    fn new(line: Line, columns: Column) -> LineDamage {
+        LineDamage{
+            left: columns+1,
+            right: Column(0),
+            line,
+        }
+    }
+
+    #[inline(always)]
+    pub fn reset(&mut self, columns: Column) {
+        self.left = columns+1;
+        self.right = Column(0);
+    }
+}
+
+struct Damage {
+    line_damage: Vec<LineDamage>,
+    full_damage: bool,
+    columns: Column,
```

Comment:
It should be `end_col`(or whatever name we use in our codebase for it), right? `columns` doesn't make much, because of `Column` type, so it is very confusing.

Reply:
This type is used for counting columns, e.g. `Grid::num_cols`. `end_col` implies coordinate, but `num_cols` like the Grid method could work.

Reply:
I do understand it, just lets just num_cols then.